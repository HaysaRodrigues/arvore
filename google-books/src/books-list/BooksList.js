import React from 'react';
import axios from 'axios';
import BookCard from "../book-card/BookCard";

const URL = 'https://www.googleapis.com/books/v1/volumes?q=Query&startIndex=0&maxResults=20';
export default class BooksList extends React.Component {

    constructor(props) {
        super(props)
        this.state = {
            books: []
        }
    }

    async componentDidMount() {
        const response = await axios.get(URL);
        const { data } = response;
        this.setState({books: data.items});
    }

    render() {
        return (
            <>
                {this.state.books.map((book, index) => (
                    <BookCard
                        key={book.id}
                        image={book.volumeInfo.imageLinks.smallThumbnail}
                        bookTitle={book.volumeInfo.title}/>
                ))}
            </>
        )
    }
}
