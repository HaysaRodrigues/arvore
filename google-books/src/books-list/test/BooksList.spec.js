import React from 'react'
import renderer from 'react-test-renderer';
import BooksList from "../BooksList";

describe('<BooksList/>', () => {
    test('should render book list', () => {
        const tree = renderer.create(<BooksList/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})
