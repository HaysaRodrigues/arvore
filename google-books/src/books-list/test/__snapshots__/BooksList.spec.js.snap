// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BooksList/> should render book list 1`] = `null`;
